Elemental/Components/AeTable/GenericComparer.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+public class GenericComparer<T> : IComparer<T>
+{
+    private Comparison<T> comparison;
+
+    public GenericComparer(Comparison<T> comparison)
+    {
+        if (comparison == null)
+            throw new ArgumentNullException("comparison");
```

Comment:
`throw new ArgumentNullException(nameof(comparison));` ☺

`nameof` is a neat little trick for throwing null argument exceptions